### PR TITLE
[TTAHUB-3530] Activity Report FE changes for monitoring

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This hooks up the monitoring goal to the standard goal drop down and adds the citations drop down to the objective.

Note: This does NOT include the various alerts on the FE for Grants under monitoring missing monitoring goals that's a later ticket.

## How to test

In order to setup the goals and the retrieval of Citations you will need to complete the following steps.

- Restore a prod db to have access to the monitoring data (run db:migrate)
- Open createMonitoringGoals.js and set the cutOffDate var to a date in the past like '2024-10-01'
- Run the Create Monitor Goal job via the CLI as _yarn docker:yarn:be createMonitoringGoalsCLI:local_ (or the local way)
- Ensure you now have monitoring goals in the goals table.
- Open src\services\Citation.js and make sure the cutoff date var coresponds with what you set in the job ( '2024-10-01').
- Find a recipient that is linked to a monitoring grant using the below query (from @thewatermethod).
- Create an AR, be sure to select the monitoring grant AND set a start date before going to the goals and objectives page.
- Check the 'Use standard goals' check box on the goals and objectives page and search for the goal that starts with "(Monitoring)".
- When you add a objective to that goal you should now have the ability to select citaions.
- Test various combinations of Grants that require monitoring with ones that dont. Make sure we only create monitoring goals accordingly.
- Test Creating and updating the goal fields, reviewing the report, submitting, etc.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-0

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js [hunk 1]
```diff
@@ -105,10 +111,50 @@ const GoalPicker = ({
...
     }
-
     fetchTopics();
   }, []);
 
+  // Fetch citations for the goal if the source is CLASS or RANs.
+  useEffect(() => {
+    async function fetchCitations() {
+      // If its a monitoring goal and the source is CLASS or RANs, fetch the citations.
+      if (goalForEditing && goalForEditing.source === 'Federal monitoring issues, including CLASS and RANs') {
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js:121]: any goal could have this source, right? can we use it like this?

**Comment by thewatermethod** [comment]
Running the monitoring goals script in docker:
```yarn docker:yarn:be createMonitoringGoalsCLI:local```

**Comment by thewatermethod** [comment]
query to find recipients with monitoring goals to test
```sql
SELECT r.name FROM "Grants" gr 
    INNER JOIN "Goals" go ON go."grantId" = gr.id
    INNER JOIN "Recipients" r ON gr."recipientId" = r.id
    WHERE go."createdVia" = 'monitoring'
    AND gr."regionId" = 1;
```

**Comment by thewatermethod** [comment]
When I add more recipients to a report, the monitoring goal is created for them regardless of whether or not they had it originally

**Comment by thewatermethod** [comment]
Citations don't clear out when objectives are switched (pick the monitoring goal, add an objective, switch goals, pick remove objective from the dropdown, switch back to monitoring goal, pick create a new objective. Citation is already selected)

**Comment by thewatermethod** [comment]
I would expect to see my citation selection here:

Screenshot 2024-12-06 at 10 04 40 AM

**Comment by AdamAdHocTeam** [comment]
> I would expect to see my citation selection here:
> 
> Screenshot 2024-12-06 at 10 04 40 AM

Well, we just want to world don't we lol

File: frontend/src/components/selectOptionsReset.js [hunk 2]
```diff
@@ -13,6 +14,15 @@ const selectOptionsReset = {
       height: 'auto',
     };
   },
+  groupHeading: (provided) => ({
+    ...provided,
+    fontWeight: 'bold',
+    fontFamily: 'SourceSansPro',
```
Comment (thewatermethod) [inline frontend/src/components/selectOptionsReset.js:20]: this isn't being applied, I say just remove it

File: frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js]: updated to standard.

File: frontend/src/components/selectOptionsReset.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/selectOptionsReset.js:20]: Its needed to bold the drop down label.

**Comment by AdamAdHocTeam** [comment]
> When I add more recipients to a report, the monitoring goal is created for them regardless of whether or not they had it originally

This should now be fixed.

**Comment by AdamAdHocTeam** [comment]
> Citations don't clear out when objectives are switched (pick the monitoring goal, add an objective, switch goals, pick remove objective from the dropdown, switch back to monitoring goal, pick create a new objective. Citation is already selected)

This seems to be a bug for Topics also will see what I can do.

File: frontend/src/components/selectOptionsReset.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/selectOptionsReset.js]: removed.

File: frontend/src/fetchers/citations.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable import/prefer-default-export */
+import join from 'url-join';
+import {
+  get,
+} from './index';
+
+export async function fetchCitationsByGrant(region, grantIds, reportStartDate) {
+  const formattedDate = new Date(reportStartDate).toISOString().split('T')[0];
```
Comment (thewatermethod) [inline frontend/src/fetchers/citations.js:8]: are we formatting this into the format it was already in?

File: frontend/src/fetchers/citations.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  const url = join(
+    '/',
+    'api',
+    'citations',
+    'region',
+    String(region),
+    `?grantIds=${grantIds.join('&')}&reportStartDate=${formattedDate}`,
```
Comment (thewatermethod) [inline frontend/src/fetchers/citations.js:15]:
```suggestion
    `?grantIds=${grantIds.join('&grantIds=')}&reportStartDate=${formattedDate}`,
```

File: frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js [hunk 5]
```diff
@@ -105,10 +111,54 @@ const GoalPicker = ({
       const topics = await getTopics();
       setTopicOptions(topics);
     }
-
     fetchTopics();
   }, []);
 
+  // Fetch citations for the goal if the source is CLASS or RANs.
+  useEffect(() => {
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js:118]:
```suggestion
  useDeepCompareEffect(() => {
```

**Comment by thewatermethod** [comment]
Screenshot 2024-12-09 at 10 37 28 AM

I think that goal source should be read only for monitoring goal

File: frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js (hunk 5 above)
Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js]: you'll need to import that lib as well @AdamAdHocTeam

File: frontend/src/fetchers/citations.js (hunk 3 above)
Comment (AdamAdHocTeam) [reply frontend/src/fetchers/citations.js:8]: We need YYYY-MM-DD see before and after:

image

File: frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js (hunk 5 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js]: done.

File: frontend/src/fetchers/citations.js (hunk 4 above)
Comment (AdamAdHocTeam) [reply frontend/src/fetchers/citations.js]: updated.

File: frontend/src/fetchers/citations.js (hunk 3 above)
Comment (thewatermethod) [reply frontend/src/fetchers/citations.js:8]: oh ok

**Comment by thewatermethod** [comment]
It's looking pretty good - I think fix the test and fix the source as noted in my comment above and we can probably move this forward

**Review by thewatermethod — APPROVED** [review]

**Comment by thewatermethod** [comment]
once CI passes, let's figure out how we are going to get it deployed somewhere for testing

**Pull request merged**